workerpool.go

```diff
@@ -110,6 +115,18 @@ func (p *WorkerPool) Submit(task func()) {
 	}
 }
 
+// TrySubmit tries to enqueue a function for a worker to execute.
+// It will return ErrWorkerStopped if the worker pool has been stopped.
+//
+// Refer to Submit for more information.
+func (p *WorkerPool) TrySubmit(task func()) error {
+	if p.Stopped() {
+		return ErrWorkerStopped
+	}
```

Comment:
Seems there is a race here: If WorkerPool is stopped after the `p.Stopped` check, but before the call to `p.Submit`. Consider using `recover` and checking for a write to a closed channel error, and then returning `ErrWorkerStopped`.